extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -554,3 +561,16 @@ func (a *genericActuator) cleanupRollingUpdateCondition(ctx context.Context, wor
 	}
 	return nil
 }
+
+// getNumberOfExistingMachines gets the total number of machines existing across all machine deployments
+func (a *genericActuator) getNumberOfExistingMachines(ctx context.Context, namespace string) (int32, error) {
+	existingMachineDeployments := &machinev1alpha1.MachineDeploymentList{}
+	if err := a.client.List(ctx, existingMachineDeployments, client.InNamespace(namespace)); err != nil {
+		return 0, err
```

Comment:
isn't this fetching from api server? wouldn't fetching from cache be better?

Reply:
client provided by the controller-runtime will fetch it from the cache. Prior to controller-runtime it was required to explicitly create a lister and an informer but this is no longer required.

Reply:
okay thanks for confirming